### DISCUSSION TITLE
How to compare sources of two versions?

### DISCUSSION THREAD

**Discussion by ilkant** [Stellarium/stellarium#2827] [Q&A]
I have a problem with display and AppImage. Now I compiled Stellarium 1.1.1 on my Kubuntu 22.04. It has same problem than AppImage Stellarium. The feature become after version 1.0 RC3 and before 1.1. So, is it possible to find differences of code between those versions? Especially for the screen related code?

And if I don't find the bug, is it possible to overwrite display related code from old version 1.0 RC3 ? And Stellarium works?

**Comment by ilkant** [comment]
https://munkuvat.net/kuva/stellarium-compiled.pLJN

**Comment by 10110111** [comment]
Well, if you really want to just compare code, you have two options: either using `git diff`, or a standalone `diff`/`kdiff3`/`meld`/whatever (some of these are usable via `git difftool`). If instead you want to find the commit that broke things for you, you can look at `git bisect`, which makes it easier to use binary search for this. A good tutorial on using git-bisect can be found [in Wine Wiki](https://wiki.winehq.org/Regression_Testing#Running_the_bisection).

**Comment by ilkant** [comment]
No problem anymore. I changed Kubuntu to use Nvidia own closed drivers and screen does not have extra pieces on sky anymore. So, no need to compare Stellarium's code.
